components/scream/src/physics/p3/tests/p3_unit_tests.cpp

```diff
@@ -201,9 +201,353 @@ static void  cloud_water_autoconversion_unit_bfb_tests(){
+           1.074052404241E+00,  3.337000000000E+05,  2.834700000000E+06,  true,                true,
+           1.800000000000E+03,  2.000000000000E-01,  3.489010298223E+02,  2.864213065220E+02,  5.000000000000E-03,
+           7.142857142857E-05,  1.234457471636E+08,  7.894736842105E-06,  1.000000000000E+06,  1.000000000000E-04,
+           1.000000000000E+06,  7.142857142857E-05,  1.000000000000E-02},
+
+        {  8.981987218068E-18,  4.252854439942E-09,  2.951952291484E-09,  3.753734473341E+06,  2.659787744788E-03,
+           0.000000000000E+00,  4.369998191591E-08,  5.117108006312E+04,  1.426603114295E-14,  5.215270087819E-03,
+           1.988011238405E+03,  0.000000000000E+00,  0.000000000000E+00,  0.000000000000E+00,  9.024441257060E-07,
+           0.000000000000E+00,  0.000000000000E+00,  2.108282741596E-02,  0.000000000000E+00,  3.763055930270E-10,
+           1.079588975398E+00,  3.337000000000E+05,  2.834700000000E+06,  true,                true,
+           1.800000000000E+03,  2.000000000000E-01,  2.865623531734E+02,  2.856490883044E+02,  5.000000000000E-03,
+           7.857142857143E-05,  1.234468002205E+08,  8.421052631579E-06,  1.000000000000E+06,  1.000000000000E-04,
+           1.000000000000E+06,  7.857142857143E-05,  1.000000000000E-02},
+
+        {  3.794207254462E-17,  6.011535856641E-09,  1.800380240751E-09,  3.730999637824E+06,  1.029985706458E-02,
+           0.000000000000E+00,  4.611932237156E-08,  5.111245469131E+04,  4.451756901094E-14,  2.019579816585E-02,
+           1.122647135602E+03,  0.000000000000E+00,  0.000000000000E+00,  0.000000000000E+00,  1.087929574200E-06,
+           0.000000000000E+00,  0.000000000000E+00,  1.764574860854E-02,  0.000000000000E+00,  4.589050692518E-10,
+           1.085256336995E+00,  3.337000000000E+05,  2.834700000000E+06,  true,                true,
+           1.800000000000E+03,  2.000000000000E-01,  2.457038651196E+02,  2.848881406331E+02,  5.000000000000E-03,
+           8.571428571429E-05,  1.234478699837E+08,  8.947368421053E-06,  1.000000000000E+06,  1.000000000000E-04,
+           1.000000000000E+06,  8.571428571429E-05,  1.000000000000E-02},
+      };
+      
+      // Sync to device
+      KTH::view_1d<P3UpdatePrognosticIceData> pupidc_host("pupidc_host", Spack::n);
+      view_1d<P3UpdatePrognosticIceData> pupidc_device("pupidc_host", Spack::n);
+      
+      // This copy only copies the input variables.
+      std::copy(&pupidc[0], &pupidc[0] + Spack::n, pupidc_host.data());
+      Kokkos::deep_copy(pupidc_device, pupidc_host);
+
+      // Get data from fortran
+      for (Int i = 0; i < max_pack_size; ++i) {
+	update_prognostic_ice(pupidc[i]);
+      }
+
+      // This copy also copies the output from the fortran function into the host view. These values
+      // are need to check the values returned from
+      std::copy(&pupidc[0], &pupidc[0] + Spack::n, pupidc_host.data());      
```

Comment:
This copy is not needed. You can just use pupidc for validating CXX data.

Reply:
I will resolve this after making changes for "Kokkos::create_mirror_view"